artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java

```diff
@@ -52,6 +55,10 @@ public static Path prepareVersionPluginOutput() throws IOException {
         return quarkusRepoDirectory;
     }
 
+    private static String normalizeVersionForUpstream(String version) {
+        return version.replaceFirst("\\.redhat-\\d+$", "").replaceFirst("\\.temporary.*$", "");
```

Comment:
This will broke when we have non-temporary versions

Reply:
It's true, ok I will remove it, thanks.